sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/IndentSingleLineFollowingConditional.cs

```diff
@@ -0,0 +1,185 @@
+            {
+                // Squiggle - "for (...)"
+                var primaryLocation = GetLocationBetweenTokens(forStatement.ForKeyword, forStatement.CloseParenToken);
+
+                ReportIssue(context, primaryLocation, forStatement.Statement, "for");
+            }
+        }
+
+        private static void CheckForEach(SyntaxNodeAnalysisContext context)
+        {
+            var forEachStatement = (ForEachStatementSyntax)context.Node;
+            if (!IsStatementIndentationOk(forEachStatement, forEachStatement.Statement))
+            {
+                // Squiggle - "foreach (...)"
+                var primaryLocation = GetLocationBetweenTokens(forEachStatement.ForEachKeyword, forEachStatement.CloseParenToken);
+
+                ReportIssue(context, primaryLocation, forEachStatement.Statement, "foreach");
+            }
+        }
+
+        private static void CheckIf(SyntaxNodeAnalysisContext context)
+        {
+            IfStatementSyntax ifStatement = (IfStatementSyntax)context.Node;
+
+            // Special case for "else if" on the same line.
+            // In that case, we'll check that the statement is more indented then the "else", not the "if".
+            // Highlighting: "if (...)", or  "else if (...)" as appropriate
+
+            SyntaxNode controlNode;
+            SyntaxToken startToken;
+            string conditionLabelText;
+            if (ifStatement.Parent is ElseClauseSyntax elseClause
+                && ifStatement.GetLineNumberToReport() == elseClause.GetLineNumberToReport())
+            {
+                controlNode = elseClause;
+                startToken = elseClause.ElseKeyword;
+                conditionLabelText = "else if";
+            }
+            else
+            {
+                controlNode = ifStatement;
+                startToken = ifStatement.IfKeyword;
+                conditionLabelText = "if";
+            }
+
+            if (!IsStatementIndentationOk(controlNode, ifStatement.Statement))
+            {    
+                var primaryLocation = GetLocationBetweenTokens(startToken, ifStatement.CloseParenToken);
+
+                ReportIssue(context, primaryLocation, ifStatement.Statement, conditionLabelText);
+            }
+        }
+
+        private static void CheckElse(SyntaxNodeAnalysisContext context)
+        {
+            var elseClause = (ElseClauseSyntax)context.Node;
+            if (!IsStatementIndentationOk(elseClause, elseClause.Statement))
+            {
+                // Just highlight the "else" keyword
+                ReportIssue(context, elseClause.ElseKeyword.GetLocation(), elseClause.Statement, "else");
+            }
+        }
+
+        private static bool IsStatementIndentationOk(SyntaxNode controlNode, SyntaxNode conditionallyExecutedNode) =>
+            conditionallyExecutedNode is BlockSyntax ||
+            GetFirstCharPosition(conditionallyExecutedNode) > GetFirstCharPosition(controlNode);
```

Comment:
Just checking the character positions is not accurate enough if the indent characters contain a mix of tabs and non-tabs.
Currently, this is causing 6FPs in the ITs: some because Nancy uses Java-style formatting and so has lines like "[tab[tab]} else if", and some because the indent characters are not used consistently i.e. tabs on some lines and spaces on others.

I can fix all the FPs by doing a smarter comparison i.e. fetching the text strings for indents, comparing the number of tabs etc. Do you think it's worth it?

(FYI it looks like the _MultilineBlocksWithoutBraces_ rule might have the same indent-comparison issue, but I guess it's not a problem for the source code used in the ITs).

Reply:
Knowing that the IDE will usually reformat your code, this rule is not the most useful and I would go for the simplest solution and not spend too much time.